iree/compiler/Dialect/HAL/Target/LLVM/LLVMAOTTarget.cpp

```diff
@@ -285,8 +285,16 @@ class LLVMAOTTargetBackend final : public TargetBackend {
       int64_t localMemorySize = entryPointOp.workgroup_local_memory()
                                     .getValueOr(APInt(64, 0))
                                     .getSExtValue();
+      // Get source location info from GenerateDispatchTagsPass in Flow pass pipeline.
+      auto execOp = variantOp->getParentOfType<IREE::Flow::ExecutableOp>();
 
-      libraryBuilder.addExport(entryPointOp.getName(), "",
+      mlir::NamedAttrList source_loc_tags = execOp->getAttrOfType<mlir::DictionaryAttr>("source_loc_tags").getValue();
+
+      mlir::NamedAttribute sourceIRSummary = source_loc_tags.getNamed("sourceIRSummary").getValue();
+
+      llvm::StringRef source_loc_tag = std::get<1>(sourceIRSummary).cast<mlir::StringAttr>().getValue();				 
```

Comment:
Some general tips regarding segfaults:

1) Attach a debugger (we really like using debuggers within VSCode)
2) See https://mlir.llvm.org/getting_started/Debugging/ . In particular, dump output before/after passes you're working with and run with `-debug` where relevant
3) Add null checks for any MLIR attribute queries, casts, etc. many properties are optional or polymorphic, so make sure you're operating on the data you think you are. Some random examples of the coding style: https://github.com/google/iree/blob/f74e4c3121de68baba81f1822042bdc485c02f2e/iree/compiler/Dialect/VM/IR/VMOps.cpp#L444-L450 https://github.com/google/iree/blob/f74e4c3121de68baba81f1822042bdc485c02f2e/iree/compiler/Dialect/HAL/IR/HALOpFolders.cpp#L93-L100
4) For printf-style debugging, `LLVM_DEBUG` and `llvm::dbgs()` are useful, like this: https://github.com/google/iree/blob/f74e4c3121de68baba81f1822042bdc485c02f2e/iree/compiler/Dialect/Util/Transforms/FoldGlobals.cpp#L403-L407